crates/polars-arrow/src/legacy/kernels/time.rs

```diff
@@ -70,3 +70,39 @@ pub fn convert_to_naive_local_opt(
         LocalResult::None => None,
     }
 }
+
+#[cfg(feature = "timezones")]
+pub fn naive_utc_to_naive_local_in_new_time_zone(
```

Comment:
nice descriptive names 👍 do you want to make a separate PR to make `convert_to_naive_local_opt` similarly descriptive?